design/proxy_public_images.md

```diff
@@ -0,0 +1,53 @@
+# Allow public images to be proxied through replicated registry
+
+The goal is to reduce number of domains that need to be accessible through proxies from private networks.
+
+## Goals
+
+- All images will be pulled through proxy.replicated.com or registry.replicated.com.
+- None of the other registries need to be accessible through firewalls.
+
+## Non Goals
+
+- There will be no magic anonymous access to registries via proxy.replicated.com:
+    - Vendors will still need to connect to external registry with valid credentials even if all images are public.
+    - `imagePullSecret` will be required in order to pull public images.
+- Public images will not retain original public names on the private instance.  All image names will begin with `proxy.replicated.com`.
+- This change will not apply to airgap builds and installs.  If instance has access to pubic registry, missing image from airgap bundle will still be pulled from the internet.
```

Comment:
I think this make sense, and I understand. I have a (perhaps tangential) question -- what do you expect to happen in the case that

1. `proxyPublicImages` set to true
2. kURL airgap bundle is brought onto an online node
3. kURL airgap bundle is installed with `cat install.sh | sudo bash -s airgap`
4. license is uploaded, online mode selected

In this case, public images would still get pulled from proxy.replicated.com, right?

Does this count as an "airgap install"? Or is "airgap install" only in reference to how the license is configured on initial setup?

Reply:
Yes, images will be proxied.  This still counts as an online install.  None of this is about kURL.

Reply:
cool. thank you!